### ISSUE TITLE
Customize cilium deployment

### ISSUE THREAD

**Issue by vincentmli** [aws/eks-anywhere#362]
**What happened**:
Changed cilium tunnel from `geneve` to `vxlan` by

`kubectl edit configmap -n kube-system cilium-config`

then delete cilium agent
`kubectl  delete pod -l k8s-app=cilium -A`

it appears new cilium agent did not start and kubectl lost connection to api server

root@cilium-worker:/home/vincent# kubectl  get po -l k8s-app=cilium  -A
```
Error from server (InternalError): an error on the server ("") has prevented the request from succeeding
```

I had to re-install from scratch to recover

**What you expected to happen**:

I expect users can customize their cilium deployment like changing cilium configmap `cilium-config`, change cilium daemonset to pull cilium image from different repo...etc

**How to reproduce it (as minimally and precisely as possible)**:

do as above

**Anything else we need to know?**:

**Environment**:
- EKS Anywhere Release: 0.5
- EKS Distro Release:

**Comment by vincentmli** [comment]
It would be fine like before eks anywhere deployment,  I can edit some cilium manifest yaml file or some other method to customize the cilium deployment, I don't need to customize cilium after eks anywhere deployed if that adds difficulties, my use case is mainly to debug cilium issue in eks anywhere.

**Comment by vivek-koppuru** [comment]
@vincentmli Just to confirm, what OS is this for? I believe Bottlerocket has some issues with `vxlan`.

**Comment by vincentmli** [comment]
@vivek-koppuru the VM is  ubuntu 20.04, I suspect the problem is right after I delete the cilium agent pod, the it would cause network problem cause no CNI available, due to the way the eks anywhere is deployed with kind. just my guess though. I don't have problem with kubesphere because kubesphere run an actual k8s cluster, not in kind.

**Comment by vivek-koppuru** [comment]
@vincentmli What do you mean by deployed with kind? The CNI is deployed once the Kubernetes cluster is created, so the CNI spec is being applied directly on the workload cluster right? That's interesting that an eks anywhere environment doesn't allow for this functionality, which is something that I would want to look into to understand as well.

**Comment by vincentmli** [comment]
@vivek-koppuru 
this is what I mean by `kind`
```
2021-10-01T16:33:38.160Z	V4	Creating kind cluster	{"name": "dev-cluster-eks-a-cluster", "kubeconfig": "dev-cluster/generated/dev-cluster.kind.kubeconfig"}
2021-10-01T16:33:38.161Z	V6	Executing command	{"cmd": "/usr/bin/docker run -i --network host -v /home/vincent:/home/vincent -w /home/vincent -v /var/run/docker.sock:/var/run/docker.sock --entrypoint kind public.ecr.aws/eks-anywhere/cli-tools:v0.1.0-eks-a-1 create cluster --name dev-cluster-eks-a-cluster --kubeconfig dev-cluster/generated/dev-cluster.kind.kubeconfig --image public.ecr.aws/eks-anywhere/kubernetes-sigs/kind/node:v1.21.2-eks-d-1-21-4-eks-a-1 --config dev-cluster/generated/kind_tmp.yaml"}
Creating cluster "dev-cluster-eks-a-cluster" ...
 • Ensuring node image (public.ecr.aws/eks-anywhere/kubernetes-sigs/kind/node:v1.21.2-eks-d-1-21-4-eks-a-1) 🖼  ...
 ✓ Ensuring node image (public.ecr.aws/eks-anywhere/kubernetes-sigs/kind/node:v1.21.2-eks-d-1-21-4-eks-a-1) 🖼
```

there could be issue between cilium and kind. 
I even tried to use cilium cli command to upgrade the cilium without manually editing cilium daemonsets, still same problem 
```
root@cilium-worker:/home/vincent# cilium upgrade dev-cluster-eks-a-cluster
🚀 Upgrading cilium-operator to version quay.io/cilium/operator-generic:v1.10.4...
🚀 Upgrading cilium to version quay.io/cilium/cilium:v1.10.4...
⌛ Waiting for Cilium to be upgraded...

```

from another ssh session, no more API response from kubectl

```
root@cilium-worker:~# kubectl  get po -o wide -A
Error from server (InternalError): an error on the server ("") has prevented the request from succeeding (get pods)
root@cilium-worker:~# echo $KUBECONFIG
/home/vincent/dev-cluster/dev-cluster-eks-a-cluster.kubeconfig
```

so back to my original request, is it possible to customize cilium deployment like versions, configmap cilium-config before deploy eks anywhere, we don't need to figure out what's wrong after eks anywhere deployment, that maybe too much to do and too difficult to trouble shoot

**Comment by vivek-koppuru** [comment]
@vincentmli We currently don't have the ability to make specific changes to the cilium config before deploying eks anywhere. In terms of kind, cilium being deployed on the kind cluster is independent of what is being deployed on the eks-a cluster (as kind is just a means to create the eks-a cluster). Since all we do is install cilium, I am not sure what happens if we change it, but I see that you opened an issue on there as well.

**Comment by vincentmli** [comment]
@vivek-koppuru right, what I am asking is to improve eks anywhere to allow customize cilium deployment, simple as able to select cilium versions, is that difficult task to do ? :)

**Comment by vivek-koppuru** [comment]
@vincentmli Got it, I want to understand whether it's simply something where users can specify the version of cilium they want, or change more configuration like geneva to vxlan that you were trying? The reason why we don't support this currently is because with EKS-A, we wanted to abstract some of this dependencies a little for user to have a controlled set of dependencies deployed. In terms of what you are trying to do, based on what I understand, you want this ability to be able to debug a cilium issue on EKS-A?

**Comment by vincentmli** [comment]
@vivek-koppuru right for average users, able to select a cilium version to deploy brings  more flexibility. for my use case, I do prefer to have the capability to change cilium configmap cilium-config, or use cilium cli command, or helm, whatever available to change some cilium configuration settings before the deployment

**Comment by vivek-koppuru** [comment]
@vincentmli Sorry for the delay, but I created this issue to track interest in this functionality. We don't have plans for this as of now but based on other use cases and interest, as well as our roadmap, we will explore this option more.

https://github.com/aws/eks-anywhere/issues/393

**Comment by TerryHowe** [comment]
I'm going to close this ticket out in favor of #393 which can be used to track interest in different Cilium configuration. I think we are up to v1.10.11 now and as I recall, we only support geneve
